### ISSUE TITLE
As a content admin, I want images from the Bodleian Genizah collection made available as IIIF so they can be displayed on the site and be linked to transcription text.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/geniza#1061]
## dev notes
- [x] write a script to parse Bodleian Genizah TEI/XML and generate iiif manifests
   - [x] add logo
   - [x] add terms of use / rights statement
   - [x] add link to their display of the object
   - [x] update geniza templates as necessary for display
- [x] download/gather Bodleian original images
- [x] write a script to generate iiif image api level 0 tiles for all original images
- [x] figure out how to generate the image tiles
- [x] figure out up github repo setup to serve out static iiif via github pages
- ~write a script to generate a iiif collection manifest to aggregate the manifests~ skip, not essential
- [x] write a script to generate a csv file of shelfmarks and urls for import into PGP application
- [x] add readme to bodleian iiif github repo documenting how it was generated / decisions made / rights, etc.

### setup repos for each series, publish, and import into production:
- [x] create template repository and document steps in script readme
- [x] setup static iiif for Bodleian MS Heb a
- [x] clean up static iiif repositories & template

### iiif redux
- [x] work with PUL to get bodleian images into new serverless iiif instance
- [x] revise bodleian script to generate manifests that reference images in the PUL server
- [x] add bodleian manifests to a new folder in the genizalab iiif repo and publish
- [x] generate csv for import into production

**Comment by rlskoeser** [comment]
@mrustow @kseniaryzhova I have a subset of Bodleian iiif ready for review on the test site!  I've generated manifests and image tiles for all of the MS Heb a. series (volume? collection?), and have loaded them on the test site.

Here's a link to a public search for all of the documents that reference this series:
https://test-geniza.cdh.princeton.edu/en/documents/?q=shelfmark%3A%22bodl+ms+heb+a%22

things to check:
- [ ] images display in search
- [ ] images display on document detail page
- [ ] images are zoomable
- [ ] image permissions statement displays attribution information and links to terms of use
- [ ] image permissions section displays Bodleian logo
- [ ] image thumbnails display in admin document edit page

In case you are interested, here's the repo where I'm managing and publishing the iiif content (I think that I will rename the `-abc` to just `-a` and create one repo for each volume, since the size makes it slow to work with and publish). I still need to add a readme documenting what this is, how it was created, permissions, etc.; planning for this to be the model for how I structure all of them:
https://github.com/princetongenizalab/iiif-bodleian-abc

**Comment by kseniaryzhova** [comment]
@rlskoeser it all works for me! Closing!

**Comment by rlskoeser** [comment]
@kseniaryzhova thanks for reviewing! Glad to hear everything is working. I'm going to re-open because I haven't completed all the steps that I need to do to finalize this, but I'm very relieved to hear all the tests passed for you.

**Comment by rlskoeser** [comment]
@mrustow would you review the language to let me know if you'd like to see any changes before I start generating and publishing these in the final version?

I copied the Bodleian image permissions statement from a different manifest; you can see that in the image & permissions statement under the images in PGP, e.g.:
https://test-geniza.cdh.princeton.edu/en/documents/6318/

I put a "Provenance" statement into the manifest metadata to indicate the origin of the IIIF (although now I think about it maybe that doesn't make sense — provenance is usually for the materials, not the iiif). Here's the text I used:
> Static IIIF generated by Center for Digital Humanities and Princeton and Princeton Geniza Lab from Bodleian Libraries, University of Oxford. 

We're not displaying that text on the PGP site. If we want it displayed, easiest thing would be to add it into the rights statement (IDK if it needs to be shown or not; I don't feel strongly one way or the other).

If you want to see how the generated  manifest looks in a standard iiiif viewer, you can load one in a Mirador demo:
Mirador demo url: https://mirador-dev.netlify.app/__tests__/integration/mirador/
sample manifest url: https://princetongenizalab.github.io/iiif-bodleian-abc/manifests/ms-heb-a-2-12.json

click the blue + in the top left, then add resource; should load and be listed as the top item in the list; select to view; close other panes. Hamburger menu on the left will open the metadata viewer (info, rights).

Here's how it looks in Universal Viewer: https://uv-v4.netlify.app/#?manifest=https://princetongenizalab.github.io/iiif-bodleian-abc/manifests/ms-heb-a-2-12.json

semi-unrelated side note: would it be helpful to make the iiif manifests we're using in PGP discoverable for use elsewhere? We could add links with the iiif logo that you often see on sites that support iiif.

**Comment by mrustow** [comment]
Looks good. There's a typo here: 
<img width="224" alt="image" src="https://user-images.githubusercontent.com/73319225/189748669-f33253b3-5656-4b30-b604-b093b1262a04.png">

I would alter the text as follows:
Static IIIF generated by the Center for Digital Humanities at Princeton and the Princeton Geniza Lab from images online at the Bodleian Libraries, University of Oxford. 

And then can we we link "online" to this link? http://genizah.bodleian.ox.ac.uk

**Comment by rlskoeser** [comment]
@mrustow whoops, there is a typo! thank you for catching.

We can link to their Genizah site, but "online" isn't a very descriptive link word. Could we say "Genizah Fragments at the Bodleian Libraries" for the link text? (it looks like that is the name of the site)

Are you proposing this text to replace the attribution? I think we should still include the link to their terms of use. We don't have to include the CC-BY-NC because I've put that in separately as a license, which we're displaying.

**Comment by mrustow** [comment]
online --> genizah fragments at the Bodleian Libraries: even better! I struggled with how to phrase that!

I was proposing that text for the provenance statement. agreed on the terms of use. you're the boss of best practices, though!

**Comment by rlskoeser** [comment]
@mrustow thank you! just wanted to be clear what you were proposing. I like putting the provenance in the manifest in case someone tries to use it elsewhere, and your proposed language looks good to me. I don't think we need to display it on our site.

Appreciate the quick review. I'll incorporate the text changes and then work on getting the first set published. The files are large enough in aggregate that I'm going to have to do it in batches, but once I get the approach figured out and documented I should be able to keep doing them in chunks and adding them to the site as each set is ready.

**Comment by rlskoeser** [comment]
My static solution doesn't work with the scale of the bodleian images — we're within limits for github repos but not for github pages, and it's enough content that my approach is pretty slow and time intensive (which would be ok if it worked because the static solution is basically zero maintenance, but it won't work). Regrouping on this and conferring with PUL / Francis about iiif image server solutions. GitHub is still a good solution for serving out our custom and remixed manifests.

**Comment by mrustow** [comment]
Ooh bummer. What about using smaller versions of the Bodleian images — would that solve the problem?

On Sep 19, 2022, at 1:29 PM, Rebecca Sutton Koeser ***@***.******@***.***>> wrote:



My static solution doesn't work with the scale of the bodleian images — we're within limits for github repos but not for github pages, and it's enough content that my approach is pretty slow and time intensive (which would be ok if it worked because the static solution is basically zero maintenance, but it won't work). Regrouping on this and conferring with PUL / Francis about iiif image server solutions. GitHub is still a good solution for serving out our custom and remixed manifests.

**Comment by rlskoeser** [comment]
images are loaded to new PUL iiif image server, iiif manifests have been generated and published via github pages in the geniza lab github organization (and canvas uris were fixed to make them unique 😅 ), and a csv of manifests & shelfmarks has been generated for loading into PGP.

Tested loading them on the test site with v4.9 pre-release code and migrated transcription content — transcriptions are attaching nicely to the images, transcriptions are editable, and everything seems to be working. 

I checked the Documents associated with the previous set of images I'd imported for the Bodl. MS Heb a series, and only one of them had image order override defined. That will get lost when I import the new images, but since it's just one record I will just update the order back to what it should be.

These are ready to go. And they don't even require any code changes, so the bodleian images can be loaded _before_ the new release goes out!
